Sources/Storage.swift

```diff
@@ -8,28 +8,117 @@
 
 import Foundation
 
-open class Storage: NSObject {
+open class Storage {
+    public enum Change {
+        case appended([RequestModel])
+        case removed(indice: Range<Int>)
+        case updated(RequestModel, at: Int)
+        case cleared
+    }
+
+    public final class Token: Hashable {
+        weak var storage: Storage?
+
+        fileprivate init(storage: Storage) {
+            self.storage = storage
+        }
+
+        public func stopObservation() {
+            storage?.transaction { storage?.observers[self] = nil }
+        }
+
+        deinit {
+            stopObservation()
+        }
+
+        public static func == (lhs: Token, rhs: Token) -> Bool {
+            return lhs === rhs
```

Comment:
isn't that a recursive call?

Reply:
No.

Reply:
Ih, I misread and didn't see it was triple-equals, my bad. All good then 👍